### PR TITLE
Course user badge fixes

### PR DESCRIPTION
**Pull request by weiqingtoh**
- Fixed ugly wrapping for course user badge at around 10xx pixels. 
- Linked course user badge to course-user page (to be done).
- Shifted exp progress text into progress bar. 
- Plus a small fix in one of the old locales when I implemented `display_progress_bar`. 

Before:
screen shot 2016-02-05 at 5 25 16 pm

After: 
screen shot 2016-02-10 at 3 26 20 pm

### CODE REVIEW

File: app/views/course/users/invitations.html.slim [hunk 1]
```diff
@@ -6,7 +6,8 @@
 - accepted_course_users = @course_users.each.select { |user| !user.invited? }.size
 - progress = accepted_course_users * 100 / [@course_users.length, 1].max
 = display_progress_bar progress, [] do
-  = t('.progress', accepted: accepted_course_users, total: @course_users.size)
+  = t('course.users.invitations.progress', accepted: accepted_course_users,
```
Comment (lowjoel) [inline app/views/course/users/invitations.html.slim]: This shorthand should work... what was wrong with it?

Comment (lowjoel) [inline app/views/course/users/invitations.html.slim]: Also did you see that the invitations page' exp progress bar follows the size of the invitations bar?

Comment (weiqingtoh) [inline app/views/course/users/invitations.html.slim]: Since `display_progress_bar` renders a partial, the locale source was different - ie. `en.layouts.progress_bar.progress` if we use `.progress`. Discovered this when I was adding text to the course user badge as well. 

By default this should not be the case, but not sure if there is a way around this though? I could construct the entire `display_progress_bar` within the helper method itself >_<

File: app/assets/stylesheets/course/layout.scss [hunk 2]
```diff
@@ -3,20 +3,18 @@
...
       margin-bottom: 10px;
       margin-top: 10px;
-    }
-  }
 
-  #course-user-achievements {
-    h5,
-    p {
-      text-align: center;
+      .course-user-exp {
```
Comment (lowjoel) [inline app/assets/stylesheets/course/layout.scss]: remember what we said about contractions? :smile:

File: app/helpers/application_widgets_helper.rb [hunk 3]
```diff
@@ -90,6 +90,9 @@ def delete_button(name, options = nil, html_options = nil, &block)
   # Display a progress_bar with the given percentage and styling. The percentage is assumed to
   # be a number ranging from 0-100. In addition, a block can be passed to add custom text.
   #
+  # Note: I18n translations used in the block should state the full localization path. This
+  # is because a layout is rendered for this method.
```
Comment (lowjoel) [inline app/helpers/application_widgets_helper.rb]: one way around this is to instance_exec within the block. do you want to have a go in a separate PR?

File: app/views/course/users/invitations.html.slim [hunk 4]
```diff
@@ -5,8 +5,9 @@
 
 - accepted_course_users = @course_users.each.select { |user| !user.invited? }.size
 - progress = accepted_course_users * 100 / [@course_users.length, 1].max
-= display_progress_bar progress, [] do
-  = t('.progress', accepted: accepted_course_users, total: @course_users.size)
+= display_progress_bar progress, ['course-user-registration'] do
```
Comment (lowjoel) [inline app/views/course/users/invitations.html.slim]: I think we should use display_progress_bar() because this isn't a DSL call

File: app/views/layouts/_course_user_badge.html.slim [hunk 5]
```diff
@@ -9,6 +11,6 @@ div#course-user-badge
         p = course_user.achievement_count
   div.col-xs-6#course-user-level
     - unless controller.current_component_host[:course_levels_component].nil?
-      / TODO: Link to the course_user course summary page
-      h5 = t('.levels') + ' ' + course_user.level_number.to_s
-      p #{course_user.experience_points} / #{course_user.next_level_threshold} EXP
+      = link_to_course_user course_user do
```
Comment (lowjoel) [inline app/views/layouts/_course_user_badge.html.slim]: same here, ()

**Comment by weiqingtoh** [comment]
Fixed the other comments for now. 

> one way around this is to instance_exec within the block. do you want to have a go in a separate PR?
Will try this - spent a couple of hours trying to figure it out but got nowhere, will need to double check how to do it.

**Comment by lowjoel** [comment]
@weiqingtoh https://cmii.estl.moe/courses/2/invitations it seems to be fixed with #798, so maybe the first commit can be removed?

I am also thinking of applying this patch under your 2nd commit:

```diff
diff --git a/app/assets/stylesheets/course/layout.scss b/app/assets/stylesheets/course/layout.scss
index 70c8ceb..bd1b277 100644
--- a/app/assets/stylesheets/course/layout.scss
+++ b/app/assets/stylesheets/course/layout.scss
@@ -1,22 +1,24 @@
-#course-user-badge {
-  #course-user-progress {
-    .progress {
-      margin-bottom: 10px;
-      margin-top: 10px;
+.course-layout {
+  #course-user-badge {
+    #course-user-progress {
+      .progress {
+        margin-bottom: 10px;
+        margin-top: 10px;
+      }
     }
-  }
 
-  #course-user-achievements {
-    h5,
-    p {
-      text-align: center;
+    #course-user-achievements {
+      h5,
+      p {
+        text-align: center;
+      }
     }
-  }
 
-  #course-user-level {
-    h5,
-    p {
-      text-align: right;
+    #course-user-level {
+      h5,
+      p {
+        text-align: right;
+      }
     }
   }
 }
diff --git a/app/views/layouts/course.html.slim b/app/views/layouts/course.html.slim
index 0d48df9..a31b29e 100644
--- a/app/views/layouts/course.html.slim
+++ b/app/views/layouts/course.html.slim
@@ -1,7 +1,7 @@
 - layout = controller.parent_layout(of_layout: 'course') || controller.current_layout
 = render within_layout: layout do
   = render_breadcrumbs
-  div.row
+  div.row.course-layout
     div.col-lg-2.col-md-3.col-sm-4
       div.container-fluid
         div.row
```

File: app/views/layouts/_course_user_badge.html.slim [hunk 6]
```diff
@@ -9,6 +12,6 @@ div#course-user-badge
         p = course_user.achievement_count
   div.col-xs-6#course-user-level
     - unless controller.current_component_host[:course_levels_component].nil?
-      / TODO: Link to the course_user course summary page
-      h5 = t('.levels') + ' ' + course_user.level_number.to_s
-      p #{course_user.experience_points} / #{course_user.next_level_threshold} EXP
+      = link_to_course_user(course_user) do
+        h5 = t('.levels')
+        p = course_user.level_number.to_s
```
Comment (lowjoel) [inline app/views/layouts/_course_user_badge.html.slim]: honestly the to_s is redundant, but I'll still merge this.

File: app/views/layouts/course.html.slim [hunk 7]
```diff
@@ -1,13 +1,12 @@
 - layout = controller.parent_layout(of_layout: 'course') || controller.current_layout
 = render within_layout: layout do
   = render_breadcrumbs
-  div.row
+  div.row.course-layout
```
Comment (lowjoel) [inline app/views/layouts/course.html.slim]: So you applied the diff I commented with? The CSS would need to be updated also.

Comment (weiqingtoh) [inline app/views/layouts/course.html.slim]: yup .course-layout was added?
